addons/test_mail/tests/test_message_post.py

```diff
@@ -36,6 +41,19 @@ def test_message_post_return(self):
         messageId = call_kw(self.env['mail.channel'], 'message_post', [test_channel.id], {'body': 'test'})
         self.assertTrue(isinstance(messageId, int))
 
+    @users('demo')
+    def test_mail_add_signature(self):
+        user = self.env.user
+        self.test_track.user_id = user
+
+        msg_true = self.test_track.message_post(body="Test body", mail_auto_delete=False, add_sign=True, partner_ids=[self.partner_1.id, self.partner_2.id], email_layout_xmlid="mail.mail_notification_paynow")
```

Comment:
When using mail classes, you have mock_mail_gateway notably that is available (and other wrappers), allowing to automatically wrap and mock what mail is doing. Notably you have access to self._new_mails which contains mail.mail created under the wrapper, allowing to check that 1 email is prepared, and you don't have to search on it, risking any false positi